enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,891 @@
+The transitional state will be `Migrating` again but the sync controller will wait for the Cluster API controllers to acknowledge
+the change before allowing the Machine API controllers take over management of the resource.
+In both cases, the combination of the `Migrating` status value and the spec `MachineAPI`/`ClusterAPI` value allows the migration controller
+to determine which direction the intended migration is currently taking.
+
+When the conversion cannot proceed for some reason, for example, a feature is in use in Machine API that is not present in Cluster API, the following will happen:
+1. Cluster admin identifies a MachineSet to migrate to Cluster API
+1. Cluster admin uses `oc edit` to update the value of the `spec.authoritativeAPI` field to `ClusterAPI`
+1. The migration controller determines that the `Synchronized` condition is set to `False`, showing a long standing conversion error
+1. The `status.authoritativeAPI` field is not updated and the `status.synchronizedGeneration` is not updated
+1. The Machine API controllers continue to manage the MachineSet.
+1. Where appropriate, the cluster admin contacts support for additional help/timelines for availability of the missing feature
+
+Long standing syhcnronization errors, where the admin has requested a transition by changing the `authoritativeAPI`, will result in alerts firing to indicate to the
+cluster admin that the transition is not occurring as they requested.
+
+#### Workflow extension
+
+When migration between the two authoritative API versions is not possible, the sync controller will add a condition to the resource to indicate the reason for the failure.
+However, with the above described workflow, the user is not presented with a synchronous response to the migration request.
+
+To improve the feedback loop, a webhook will be introduced to validate the migration request.
+
+The migration request will always be persisted, however, the webhook will check for the presence of the sync controller's failure condition, and return a warning response to the user if the migration is not possible.
+
+The users intention to migrate will be persisted, and the sync controller will continue to attempt to migrate the resource until the condition is resolved, but, the user is also informed of the reason for the failure synchronously.
+
+This extension is not required for the minimum viable version of this project, but, will be added in a future iteration, prior to the GA release.
+
+If the migration request is not reversed within some reasonable time frame, and alert will be fired to ensure the user has visibility that their request was not fulfilled.
+
+### API Extensions
+
+This enhancement introduces a new `authoritativeAPI` field on `spec` and `status` of Machine API resources.
+Users will set the value of this field to determine which of the Machine API or Cluster API controllers will own the resource.
```

Comment:
I assume the default value here will be `MachineAPI` if the user doesn't specify anything?

Reply:
Ah, the default is stated further down.

Reply:
That's correct, can see that in the API definitely itself